providers/PrivateMessaing/EasyAbp.NotificationService.Provider.PrivateMessaging/EasyAbp/NotificationService/Provider/PrivateMessaging/PrivateMessageNotificationSendingJob.cs

```diff
@@ -0,0 +1,69 @@
+﻿using EasyAbp.NotificationService.NotificationInfos;
+using EasyAbp.NotificationService.Notifications;
+using EasyAbp.PrivateMessaging.PrivateMessages;
+using JetBrains.Annotations;
+using System;
+using System.Threading.Tasks;
+using Volo.Abp.BackgroundJobs;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Timing;
+using Volo.Abp.Uow;
+
+namespace EasyAbp.NotificationService.Provider.PrivateMessaging
+{
+    public class PrivateMessageNotificationSendingJob : IAsyncBackgroundJob<PrivateMessageNotificationSendingJobArgs>, ITransientDependency
+    {
+        private readonly INotificationInfoRepository _notificationInfoRepository;
+        private readonly IUserUserNameProvider _userUserNameProvider;
+        private readonly INotificationRepository _notificationRepository;
+        private readonly IPrivateMessageAppService _privateMessageAppService;
+        private readonly IClock _clock;
+
+        public PrivateMessageNotificationSendingJob(
+            INotificationInfoRepository notificationInfoRepository,
+            IUserUserNameProvider userUserNameProvider,
+            INotificationRepository notificationRepository,
+            IPrivateMessageAppService privateMessageAppService,
+        IClock clock)
+        {
+            _notificationInfoRepository = notificationInfoRepository;
+            _userUserNameProvider = userUserNameProvider;
+            _notificationRepository = notificationRepository;
+            _privateMessageAppService = privateMessageAppService;
+            _clock = clock;
+        }
+
+        [UnitOfWork]
+        public virtual async Task ExecuteAsync(PrivateMessageNotificationSendingJobArgs args)
+        {
+            var notification = await _notificationRepository.GetAsync(args.NotificationId);
+
+            var userName = await _userUserNameProvider.GetAsync(notification.UserId);
+
+            if (userName.IsNullOrWhiteSpace())
+            {
+                await SaveNotificationResultAsync(notification, false, NotificationConsts.FailureReasons.ReceiverInfoNotFound);
+                return;
+            }
+
+            var notificationInfo = await _notificationInfoRepository.GetAsync(notification.NotificationInfoId);
+
+            await _privateMessageAppService.CreateAsync(new EasyAbp.PrivateMessaging.PrivateMessages.Dtos.CreateUpdatePrivateMessageDto
```

Comment:
Application service cannot be used here.

I have updated the PM module:
https://github.com/EasyAbp/PrivateMessaging/commit/e84cd76bf4f08cba0d804e03ce37daa184702d10

Now you can try to publish a [SendPrivateMessageEto](https://github.com/EasyAbp/PrivateMessaging/blob/e84cd76bf4f08cba0d804e03ce37daa184702d10/src/EasyAbp.PrivateMessaging.Domain.Shared/PrivateMessages/SendPrivateMessageEto.cs) to create a message.